presto-hive/src/test/java/com/facebook/presto/hive/parquet/write/TestDataWritableWriter.java

```diff
@@ -0,0 +1,378 @@
+            String keyName = keyType.getName();
+            ObjectInspector keyInspector = inspector.getMapKeyObjectInspector();
+
+            Type valuetype = repeatedType.getType(1);
+            String valueName = valuetype.getName();
+            ObjectInspector valueInspector = inspector.getMapValueObjectInspector();
+
+            for (Map.Entry<?, ?> keyValue : mapValues.entrySet()) {
+                recordConsumer.startGroup();
+                if (keyValue != null) {
+                    // write key element
+                    Object keyElement = keyValue.getKey();
+                    recordConsumer.startField(keyName, 0);
+                    writeValue(keyElement, keyInspector, keyType);
+                    recordConsumer.endField(keyName, 0);
+
+                    // write value element
+                    Object valueElement = keyValue.getValue();
+                    if (valueElement != null) {
+                        recordConsumer.startField(valueName, 1);
+                        writeValue(valueElement, valueInspector, valuetype);
+                        recordConsumer.endField(valueName, 1);
+                    }
+                }
+                recordConsumer.endGroup();
+            }
+
+            recordConsumer.endField(repeatedType.getName(), 0);
+        }
+        recordConsumer.endGroup();
+    }
+
+    /**
+     * It writes the primitive value to the Parquet RecordConsumer.
+     *
+     * @param value The object that contains the primitive value.
+     * @param inspector The object inspector used to get the correct value type.
+     */
+    private void writePrimitive(final Object value, final PrimitiveObjectInspector inspector)
+    {
+        if (value == null) {
+            return;
+        }
+
+        switch (inspector.getPrimitiveCategory()) {
+            case VOID:
+                return;
+            case DOUBLE:
+                recordConsumer.addDouble(((DoubleObjectInspector) inspector).get(value));
+                break;
+            case BOOLEAN:
+                recordConsumer.addBoolean(((BooleanObjectInspector) inspector).get(value));
+                break;
+            case FLOAT:
+                recordConsumer.addFloat(((FloatObjectInspector) inspector).get(value));
+                break;
+            case BYTE:
+                recordConsumer.addInteger(((ByteObjectInspector) inspector).get(value));
+                break;
+            case INT:
+                recordConsumer.addInteger(((IntObjectInspector) inspector).get(value));
+                break;
+            case LONG:
+                recordConsumer.addLong(((LongObjectInspector) inspector).get(value));
+                break;
+            case SHORT:
+                recordConsumer.addInteger(((ShortObjectInspector) inspector).get(value));
+                break;
+            case STRING:
+                String v = ((StringObjectInspector) inspector).getPrimitiveJavaObject(value);
```

Comment:
`v` -> `value` (in general we don't use abbreviations or single letters for variable names)

Reply:
the same as above, need your opinion regarding updating copy of Hive Parquet writer